scenario_player/main.py

```diff
@@ -42,9 +42,46 @@
 TRANSACTION_DEFAULTS["gas"] = lambda web3, tx: web3.eth.estimateGas(tx) * 2
 
 
+def construct_log_file_name(sub_command, data_path, scenario_fpath: Path=None)-> str:
+    file_name = f"scenario-player-{sub_command}_{datetime.now():%Y-%m-%dT%H:%M:%S}.log"
+    directory = data_path
+    if scenario_fpath:
+        directory.joinpath(scenario_fpath.stem)
```

Comment:
`Path.joinpath()` doesn't modify the Path in place, it returns a new one instead